src/write_gcm.c

```diff
@@ -3840,9 +3841,9 @@ static int wg_transmit_unique_segment(const wg_context_t *ctx,
       wg_log_json_message(
           ctx, "Sending JSON (CollectdTimeseriesRequest):\n%s\n", json);
 
-      int wg_result = wg_curl_get_or_post(response, sizeof(response),
-        ctx->agent_translation_service_url, json,
-        headers, STATIC_ARRAY_SIZE(headers), 0);
+      int wg_result = wg_curl_get_or_post(&response,
```

Comment:
If allocation fails, `response` will be unchanged, and will contain the value from the previous response. But maybe that's ok, because `wg_curl_get_or_post` will return a negative number in that case, so we won't touch the `response` except to free it.